### PR TITLE
[AppService] az webapp|functionapp config ssl import: Lookup key vault across resources groups in subscription and improve help and examples.

### PR DESCRIPTION
**Pull request by madsd**
**Description**  
Prior to this change, issues were raised #11867 as the command were failing if the key vault was not in the same resource group as the web app. An option existed to provide a resource id, but this was not documented enough to be discoverable. The change allows key vault to be imported across all resource groups in the subscription in context without any change, and if it is not found, a suggestion to provide the full resource id is displayed. Further a help example of using resource id is added.
There are no changes to the parameter list or breaking changes.

**Testing Guide**  
The two examples in az webapp config ssl import --help

**History Notes**  

[AppService] az webapp|functionapp config ssl import: Lookup key vault across resources groups in subscription and improve help and examples.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

**Comment by madsd** [comment]
@yonzhan @Juliehzl I can see it is in failed state due to the PR Title - I changed that, but it does not restart the build - is there a way to restart the build or do I need to push a small change?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -2136,7 +2136,26 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
         raise CLIError("'{}' app doesn't exist in resource group {}".format(name, resource_group_name))
     server_farm_id = webapp.server_farm_id
     location = webapp.location
-    kv_id = _format_key_vault_id(cmd.cli_ctx, key_vault, resource_group_name)
+    kv_id = None
+    if not is_valid_resource_id(key_vault):
+        kv_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_KEYVAULT)
+        key_vaults = kv_client.vaults.list_by_subscription()
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2142]: Maybe you can use a get method to avoid traversing all key vaults.

Comment (madsd) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:2142]: I looked at that, but get method requires a resource group to be specified, and the point is to search across resource groups.

**Comment by madsd** [comment]
@panchagnula fyi

**Comment by madsd** [comment]
@Juliehzl - are you waiting for something from me?

**Comment by qwordy** [comment]
@Juliehzl Please review.

**Comment by madsd** [comment]
> Do you have a test for key vault resource id?

@Juliehzl , the code does not change the functionality for import with resource id - but I am happy to add a test for that. I was looking to add a test with multiple resource groups. Do you know if ResourceGroupPreparer can handle that?

**Comment by Juliehzl** [comment]
@madsd You can add multiple ResourceGroupPreparer() to create multiple resource groups. Here is a sample for StorageAccountPreparer(), which is the same. https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-8-prepare-multiple-storage-accounts-for-tests

**Comment by madsd** [comment]
@Juliehzl thanks, I added two additional tests - one for kv by name in different resource group and one for kv by resource id.

**Comment by Juliehzl** [comment]
@madsd Can you take a look at CI test? There are something wrong in your test.

**Comment by madsd** [comment]
> @madsd Can you take a look at CI test? There are something wrong in your test.

It is strange, because that part I did not change - it is the import of the cert to Key Vault. I will do a bit of debugging to see if I can find the error.

**Comment by madsd** [comment]
@Juliehzl Finally the tests passed. Still don't know what caused it, but when I combined the cross RG with Resource Id, it worked.

**Pull request merged**